application/models/Interfaces/PermissionInterface.php

```diff
@@ -2,15 +2,32 @@
 
 interface PermissionInterface
 {
+    /**
+     * @return integer|null the owner id of the pobject
+     */
     public function getOwnerId();
     /**
      * @return array
      */
     public static function getPermissionData();
+    /**
+     * @return string|null the permission key used for read access
+     */
     public static function getMinimalPermissionRead();
     /**
      * @return bool
      */
     public function hasPermission($sPermission, $sCRUD = 'read', $iUserID = null);
+    /**
+     * @return mixed
+     */
     public function getPrimaryKey();
+    /**
+     * @return \CDbCriteria
+     */
+    public static function getPermissionCriteria();
+    /**
+     * @return self
+     */
+    public function withpermission();
```

Comment:
I am not too fan of this name.
Besides case needing to be reviewed. 

What kind of permissions will this filter? Read? List? Create?

Reply:
> I am not too fan of this name.

Same for me …
https://github.com/LimeSurvey/LimeSurvey/pull/2752#issuecomment-1334130648 

> scope name withpermission, maybe better withaccess ? or withminimalpermission ?

Reply:
withminimalaccess maybe ?

The user can know it exist, but we still unsure what he can do with it.

Reply:
> withminimalaccess maybe ?

Yes, `withminimalaccess ` doesn't sound too clear.
What does that scope does? Pick ups the ones for which a user has read/list access? right?

If so, then `withReadAccess()`?

Reply:
If we are setting standards, I would do something like this:

`withAccess($userId, $permType = Permissions::READ)`.

Right now, only read access is implemented, so any other permtype would return something like `WHRE 1=0`, to have the criteria to not return rows.

Maybe not the best timing for that ...

Reply:
> If we are setting standards, I would do something like this:
> 
> `withAccess($userId, $permType = Permissions::READ)`.
> 

No: because we don't check really the READ access Permission … 

1. READ access can come from plugin, not done for Survey model
2. SurveyGroup are shown in list even if user do not complete read it

it's more something link : User know this model exist and get the name/title.

Reply:
> READ access can come from plugin, not done for Survey model

Didn't have that in the radar. OK.

> it's more something link : User know this model exist and get the name/title.

Yes, I understand that's usually a list permission.

Then the method could be named `withListAccess()` ?

Reply:
> Then the method could be named `withListAccess()` ?

Maybe but this restrict to list, 
in this commit we use it for access : to return the same error if user try to set a non existing or a not accessible SurveyGroup.
It's the reason i think `withMinimalAccess` is OK too

I work on it tomorrow :)

Reply:
> in this commit we use it for access : to return the same error if user try to set a non existing or a not accessible SurveyGroup.

Well, I believe the important to set a name is not where is being used, but what it does internally.

> It's the reason i think withMinimalAccess is OK too

I don't agree. Minimal doesn't sound very clear.
But still, if you feel is OK, go.
I am just reviewing, not authoring.
I can raise concerns, but those are just to help, not for saying "this needs to be corrected" :)

Thanks!

Reply:
withListRight ?